### ISSUE TITLE
New Rule: Don't use JSON.stringify inside JSON.stringify

### ISSUE THREAD

**Issue by hvlxh** [eslint/eslint#17120]
### Rule details

using JSON.stringify inside JSON.stringify is useless the main gonna do it anyway

### Related ECMAScript feature

idk. JSON Object added in es5/es3 ig

### What type of rule is this?

Warns about a potential problem

### Example code

```js
// wrong
JSON.stringify({
  ...params,
  payload: JSON.stringify(packet.buffer.toJSON())
})

// correct
JSON.stringify({
  ...params,
  payload: packet.buffer.toJSON()
})
```

### Why should this rule be in the core instead of a plugin?

i don't know, probably there is no plugin to implement to feature i guess.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by Rec0iL99** [comment]
Hi @hvlxh, thanks for the issue. I'm not sure though, but this does not look like something that can be included in the core rules of ESLint. Maybe you can write a custom rule for your case or there could be a plugin that does the job for you. 

The @eslint/eslint-tsc folks will have a better idea.

**Comment by mdjermanovic** [comment]
I think this is more of a stylistic preference than a potential problem, so per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies/) this rule doesn't qualify for inclusion in the core because it isn't related to new ECMAScript features that reached stage 4 in the preceding 12 months. As @Rec0iL99 suggested, you can always make a custom rule.
